pathspider/plugins/ecnspider3.py

```diff
@@ -155,6 +169,47 @@ def create_observer(self):
             traceback.print_exc()
             sys.exit(-1)
 
+    def combine_flows(self, flow):
+        dip = flow['dip']
+        if dip in self.comparetab:
+            other_flow = self.comparetab.pop(dip)
+
+            # first has always ecn off, while the second has ecn on
+            flows = (flow, other_flow) if other_flow['ecnstate'] else (other_flow, flow)
+
+            tstart = min(flow['tstart'], other_flow['tstart'])
+            tstop = max(flow['tstop'], other_flow['tstop'])
+
+            if flows[0]['connstate'] and flows[1]['connstate']:
+                cond_conn = 'ecn.connectivity.works'
+            elif flows[0]['connstate'] and not flows[1]['connstate']:
+                cond_conn = 'ecn.connectivity.broken'
+            elif not flows[0]['connstate'] and not flows[1]['connstate']:
+                cond_conn = 'ecn.connectivity.transient'
+            else:
+                cond_conn = 'ecn.connectivity.offline'
+
+            if flows[1]['rev_syn_flags'] & TCP_SAEW == TCP_SAE:
+                cond_nego = 'ecn.negotiated'
+            else:
+                cond_nego = 'ecn.not_negotiated'
+
+            self.outqueue.put({
+                'sip': flow['sip'],
+                'dip': dip,
+                'dp': flow['dp'],
+                'conditions': [cond_conn, cond_nego],
+                'hostname': flow['host'],
```

Comment:
Does hostname and rank actually get recorded in the observatory? Do we want to output native schema formatted results for input to the observatory or is it worth also having this metadata for direct analysis?

It may be a better idea to store a separate table containing ranks/hostnames/resolved addresses and do cross-referencing?